File: go/0070E-爬楼梯.go
/*
 * @Author: SukiEva
 * @Date: 2021-11-27 09:05:36
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
package main

func climbStairs(n int) int {
	if n <= 2 {
		return n
	}
	var dp [3]int
	dp[1] = 1
	dp[2] = 2
	for i := 3; i <= n; i++ {
		sum := dp[1] + dp[2]
		dp[1] = dp[2]
		dp[2] = sum
	}
	return dp[2]
}

File: cpp/0070E-爬楼梯.cpp
/*
 * @Author: SukiEva
 * @Date: 2022-01-03 14:25:26
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    int climbStairs(int n) {
        if (n <= 1) return n;
        vector<int> dp(n + 1);
        dp[1] = 1;
        dp[2] = 2;
        for (int i = 3; i <= n; i++) {
            dp[i] = dp[i - 1] + dp[i - 2];
        }
        return dp[n];
    }
};